.vscode/c_cpp_properties.json

```diff
@@ -3,31 +3,31 @@
         {
             "name": "xtensa",
             "includePath": [
-                "/workspaces/Smooth/lib/smooth/include/**",
-                "/workspaces/Smooth/externals/**",
-                "/esp/esp-idf/components/**",
-                "/esp/tools/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/xtensa-esp32-elf/include/**"
+                "lib/smooth/include/**",
+                "externals/**",
+                "$IDF_PATH/components/**",
+                "$IDF_TOOLS_PATH/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/xtensa-esp32-elf/include/**"
             ],
             "intelliSenseMode": "gcc-x64",
-            "compilerPath": "/esp/tools/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc",
+            "compilerPath": "$IDF_TOOLS_PATH/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc",
             "cStandard": "c11",
             "cppStandard": "c++17",
-            "defines": [                
+            "defines": [
             ],
             "browse": {
                 "path": [
-                    "/esp/esp-idf",
-                    "/workspaces/Smooth/lib/smooth/include"
+                    "$IDF_PATH",
+                    "lib/smooth/include"
                 ]
             }
         },
         {
             "name": "host",
             "includePath": [
-                "/workspaces/Smooth/lib/smooth/include/**",
-                "/workspaces/Smooth/externals/**",
-                "/esp/esp-idf/components/**",
-                "/esp/tools/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/xtensa-esp32-elf/include/**"
+                "lib/smooth/include/**",
+                "externals/**",
+                "mock-idf/components/**",
+                "$IDF_TOOLS_PATH/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/include/**"
```

Comment:
Might not be possible, but can this be made version agnostic by use of another variable?